File: week_8_and_9/4_Ruby/fizz_buzz/my_solution.rb
# U3.W8-9: 


# I worked on this challenge [by myself, with: ].

# 2. Pseudocode



# 3. Initial Solution

=begin

def super_fizzbuzz(array)
	fizzbuzzed = []
	array.each do |i|
		if i % 15 == 0
			fizzbuzzed << "FizzBuzz"
		elsif i % 5 == 0
			fizzbuzzed << "Buzz"
		elsif i % 3 == 0
			fizzbuzzed << "Fizz"
		else
			fizzbuzzed << i
		end
	end
	return fizzbuzzed
end

=end


# 4. Refactored Solution


def super_fizzbuzz(array)
	fizzbuzzed = []
	array.each do |i|
		if i % 15 == 0
			fizzbuzzed << "FizzBuzz"
		elsif i % 5 == 0
			fizzbuzzed << "Buzz"
		elsif i % 3 == 0
			fizzbuzzed << "Fizz"
		else
			fizzbuzzed << i
		end
	end
	return fizzbuzzed
end



# 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE

p super_fizzbuzz([2,3]) == [2,"Fizz"]
p super_fizzbuzz([5,15]) == ["Buzz","FizzBuzz"]


# 5. Reflection 

File: week_8_and_9/3_JavaScript/grocery_list/my_solution.js
// U3.W8-9: 


// I worked on this challenge [by myself, with: ].

// 2. Pseudocode

// Convert the ruby code to JavaScript.

// 3. Initial Solution

/*
var grocerylist = {};
list = [];

var add = function(item) {
list.push(item);
};

var remove = function(item) {
delete list.item;
};

var display = function() {
console.log(list);
};

var empty = function() {
 list = [];
};

var item_numbers = function() {
	list.length;
};
*/

// 4. Refactored Solution

var grocerylist = {};
list = [];

var add = function(item) {
list.push(item);
};

var remove = function(item) {
delete list.item;
};

var display = function() {
console.log(list);
};

var empty = function() {
 list = [];
};

var item_numbers = function() {
	list.length;
};


// 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE

list = ["Milk", "Bread", "Fruits"];
console.log(grocerylist.item_numbers) === 3;
console.log(grocerylist.add("Cereal")) === ["Milk", "Bread", "Fruits", "Cereal"];
console.log(grocerylist.remove("Bread")) === ["Milk", "Fruits", "Cereal"];


// 5. Reflection 

// I just don't remember how to run a JavaScript file in the terminal